python/cudf/cudf/core/column/categorical.py

```diff
@@ -1067,11 +1069,18 @@ def __cuda_array_interface__(self) -> Mapping[str, Any]:
     def to_pandas(
         self, index: ColumnLike = None, nullable: bool = False, **kwargs
     ) -> pd.Series:
-        signed_dtype = min_signed_type(len(self.categories))
-        codes = self.cat().codes.astype(signed_dtype).fillna(-1).to_array()
-        categories = self.categories.to_pandas()
+
+        if self.categories.isnull().any():
+            col = self.copy(deep=True)
+            col[col.isnull()] = None
```

Comment:
It's a shame we need to make a deep copy and scatter here when really we just need to calculate a new mask. Maybe we could do something like:
```
new_mask = cudf.mask_from_bool_column(col.isnull()) # I forget the actual function for converting a boolean column into a bitmask
col = cudf.core.column.build_column(..., mask=new_mask, ...)
```

Reply:
Done 👍